packages/cli/src/commands/doctor/healthchecks/cocoaPods.js

```diff
@@ -0,0 +1,54 @@
+import execa from 'execa';
+import {isSoftwareInstalled} from '../checkInstallation';
+
+const hasSudoGranted = async () => {
+  try {
+    await execa('sudo', ['-nv']);
+
+    return true;
+  } catch (_ignored) {
+    return false;
+  }
+};
+
+// This is to show the loader but clear the `Password` prompt from the screen
+const reportLoader = async ({loader, type, userHasSudoGranted}) => {
+  process.stdout.moveCursor(0, userHasSudoGranted ? -1 : -2);
+  process.stdout.clearScreenDown();
+
+  loader[type]();
+};
+
+const cocoaPods = {
+  label: 'CocoaPods',
+  getDiagnosticsAsync: async () => ({
+    needsToBeFixed: !(await isSoftwareInstalled('pod')),
+  }),
+  runAutomaticFix: async ({loader}) => {
+    try {
+      // First attempt to install `cocoapods`
+      await execa('gem', ['install', 'cocoapods']);
```

Comment:
Should we rely on Homebrew/MacPorts to avoid having to invoke `sudo`? Or at least use them if they are installed.

Reply:
we have a helper function for that, let's use it :) https://github.com/react-native-community/cli/blob/master/packages/cli/src/tools/installPods.js